### ISSUE TITLE
Deletion of obsolate preview-folders

### ISSUE THREAD

**Issue by JSoko** [rullzer/previewgenerator#64]
What's happening with the preview of deleted files?
Do they stay in the preview folder or are they deleted by preview:pre-generate?

If they are not deleted, should the content of the previev-folder be deleted time by time and new created by running preview:generate-all?

**Comment by MichaIng** [comment]
This is actually not a question about the previewgenerator but about the IPreview system, which is part of nextcloud core.

I found the [preview Watcher class](https://github.com/nextcloud/server/blob/master/lib/private/Preview/Watcher.php) where it says: 

> * Class that will watch filesystem activity and remove previews as needed.

If you see the code, it indeed looks like it captures files database changes, and if it's about files and removing, it will get the related file id and remove the related subfolder in your `appData_<instanceID>/Previews`. So no regular manual clearing should be necessary.

The question is, if there is also some regular check for missed/lost files, like the `cleanup` command for the `oc_filecache` database (actually the file index of nextcloud). I did not find it, but it would be great if `files:cleanup` (and the related background job) would also check in in case delete the previews for the cleaned files. I will check for some issue about that and in case create one.
